File: medium/1653/main.rs
impl Solution {
    pub fn minimum_deletions(s: String) -> i32 {
        let mut res: i32 = 0;
        let n = s.chars().count();
        let s_bytes = s.as_bytes();
        let mut Delete_a = vec![0; n];
        let mut a_count = 0;
        let mut b_count = 0;

        for i in (0..n).rev() {
            Delete_a[i] = a_count;
            if s_bytes[i] == b'a' {
                a_count += 1;
            }
        }

        res = a_count;

        for i in 0..n {
            res = std::cmp::min(Delete_a[i as usize]+b_count, res);
            if s_bytes[i] == b'b' {
                b_count += 1;
            }
        }

        res
        
    }
}

File: medium/1653/main.c
#include<string.h>

int minimumDeletions(char* s) {
  int *Delete_a = (int*) malloc(strlen(s) * sizeof(int));
  int res = INT_MAX;

  for (int i = strlen(s)-1, a_count = 0; i >= 0; i--) {
    Delete_a[i] = a_count;
    if (s[i] == 'a') a_count++;
  }

  for (int i = 0, b_count = 0; i < strlen(s); i++) {
    res = fmin(Delete_a[i] + b_count, res);
    if (s[i] == 'b') b_count++;
  }
    
  free (Delete_a);

  return res;
}
